docs/src/content/docs/es/challenges/angular/4-context-outlet-typed.md

```diff
@@ -0,0 +1,46 @@
+---
+title: 🔴 ContextOutlet en forma de tipo
+description: El desafío 4 se trata de tipificar de manera fuerte las directivas de ngContextOutlet
+author: thomas-laforge
+challengeNumber: 4
+command: angular-context-outlet-type
+blogLink: https://medium.com/@thomas.laforge/ngtemplateoutlet-type-checking-5d2dcb07a2c6
+sidebar:
+  order: 201
+---
+
+## Información
+
+:::note[Nota]
+
+Las declaraciones de tipos permanecerán en ingles, asi como los nombres de los objetos y sus miembros.
+
+:::
+
+Angular ofrece la function estática [`ngTemplateContextGuard`](https://angular.io/guide/structural-directives#typing-the-directives-context) para reforzar el tipo de una directiva estructural.
+
+Sin embargo, el contexto de **NgTemplateOutlet** es de tipo **Object**. Con la ayuda de la guardia mencionada anteriormente, podemos mejorar ese comportamiento.
+
+## Declaración
+
+En este desafío, queremos aprender como reforzar el tipo de nuestro ng-template en AppComponent.
+
+Este desafío ofrece dos niveles de complejidad:
+
+### Nivel 1: una Interfase conocida
+
+A continuación tenemos estas lineas de código:
+
+!['Unkown' en 'Person'](../../../../assets/4/unknown-person.png 'Unkown' en 'Person')
+
+Como se puede observar, "name" es de tipo "any". La meta es inferir el tipo correcto.
+
+### Nivel 2: generic Interface
+
+Actualmente, tenemos estas lineas de código:
+
+!['Unkown' en 'Student'](../../../../assets/4/unknown-student.png 'Unkown' en 'Student')
```

Comment:
Looks like this path needs updating too, can we please take a look?

<img width="1155" alt="image" src="https://github.com/tomalaforge/angular-challenges/assets/62297014/ccc35095-507f-4ce4-a9ad-ec95128a4c3d">